### ISSUE TITLE
Reductive cut-elimination does not reduce all reducible cuts

### ISSUE THREAD

**Issue by jannikvierling** [gapt/gapt#693]
In the example below, there is a reducible cut in `reductiveCutElimination(p,false)`. I think we would expect this cut to be eliminated.
```
val f = hof"¬(∀x x = c0 ∧ ∃x f0(x) != f1(x))"
val Some(p) = EProver.getLKProof(f)
ReductiveCutElimination(p, false)
```

**Comment by gebner** [comment]
This is solved now, right?

**Comment by jannikvierling** [comment]
Yes!

**Comment by jannikvierling** [comment]
Fixed by PR #695
